File: ContainerWithMostWater.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length-1;

        int maxArea = 0;
        while(left < right){
            int calcArea = Math.min(height[left], height[right]) * (right-left);
            if(Math.min(height[left], height[right]) * (right-left) > maxArea){
                maxArea = calcArea;
            }
            if(height[left] < height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return maxArea;
    }
}

File: ContainerWithMostWater.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        left = 0
        right = len(height)-1
        maxval = 0
        while(left < right):
            total = min(height[left], height[right]) * (right-left)
            maxval = max(maxval, total)
            if(height[left] < height[right]):
                left += 1
            else:
                right -= 1
        return maxval
        